instrumentation/lettuce/lettuce-5.0/src/test/groovy/LettuceReactiveClientTest.groovy

```diff
@@ -287,4 +290,121 @@ class LettuceReactiveClientTest extends AgentTestRunner {
     }
   }
 
+  def "blocking subscriber"() {
+    when:
+    runUnderTrace("test-parent") {
+      reactiveCommands.set("a", "1")
+        .then(reactiveCommands.get("a")) // The get here is ending up in another trace
+        .block()
+    }
+
+    then:
+    assertTraces(1) {
+      trace(0, 3) {
+        span(0) {
+          operationName "test-parent"
+          errored false
+          tags {
+          }
+        }
+        span(1) {
+          operationName "SET"
+          spanKind CLIENT
+          errored false
+          childOf span(0)
+          tags {
+            "$Tags.DB_TYPE" "redis"
+          }
+        }
+        span(2) {
+          operationName "GET"
+          spanKind CLIENT
+          errored false
+          childOf span(0)
+          tags {
+            "$Tags.DB_TYPE" "redis"
+          }
+        }
+      }
+    }
+  }
+
+  def "async subscriber"() {
+    when:
+    runUnderTrace("test-parent") {
+      reactiveCommands.set("a", "1")
+        .then(reactiveCommands.get("a")) // The get here is ending up in another trace
+        .subscribe()
+    }
+
+    then:
+    assertTraces(1) {
+      trace(0, 3) {
+        span(0) {
+          operationName "test-parent"
+          errored false
+          tags {
+          }
+        }
+        span(1) {
+          operationName "SET"
+          spanKind CLIENT
+          errored false
+          childOf span(0)
+          tags {
+            "$Tags.DB_TYPE" "redis"
+          }
+        }
+        span(2) {
+          operationName "GET"
+          spanKind CLIENT
+          errored false
+          childOf span(0)
+          tags {
+            "$Tags.DB_TYPE" "redis"
+          }
+        }
+      }
+    }
+  }
+
+  def "async subscriber with specific thread pool"() {
+    when:
+    runUnderTrace("test-parent") {
+      reactiveCommands.set("a", "1")
+        .then(reactiveCommands.get("a")) // The get here is ending up in another trace
```

Comment:
I should have removed these comments. They no longer apply now that the tests pass